browser/ui/containers/containers_icon_generator.cc

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/containers/containers_icon_generator.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/notreached.h"
+#include "brave/components/vector_icons/vector_icons.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/image/canvas_image_source.h"
+#include "ui/gfx/image/image_skia.h"
+#include "ui/gfx/paint_vector_icon.h"
+#include "ui/gfx/vector_icon_types.h"
+
+namespace containers {
+namespace {
+
+const gfx::VectorIcon& GetVectorIconFromIconType(mojom::Icon icon) {
+  switch (icon) {
+    case mojom::Icon::kPersonal:
+      return kLeoContainerPersonalIcon;
+    case mojom::Icon::kWork:
+      return kLeoContainerWorkIcon;
+    case mojom::Icon::kShopping:
+      return kLeoContainerShoppingIcon;
+    case mojom::Icon::kSocial:
+      return kLeoContainerSocialIcon;
+    case mojom::Icon::kEvents:
+      return kLeoContainerEventsIcon;
+    case mojom::Icon::kBanking:
+      return kLeoContainerBankingIcon;
+    case mojom::Icon::kStar:
+      return kLeoContainerStarIcon;
+    case mojom::Icon::kTravel:
+      return kLeoContainerTravelIcon;
+    case mojom::Icon::kSchool:
+      return kLeoContainerSchoolIcon;
+    case mojom::Icon::kPrivate:
+      return kLeoContainerPrivateIcon;
+    case mojom::Icon::kMessaging:
+      return kLeoContainerMessagingIcon;
+  }
+
+  if (icon < mojom::Icon::kMinValue || icon >= mojom::Icon::kMaxValue) {
```

Comment:
> `icon >= mojom::Icon::kMaxValue`

I believe it should be `icon > mojom::Icon::kMaxValue`. `kMaxValue` is a valid value:
```
  kMessaging = 10,
  
  kDefault = 0,
  kMinValue = 0,
  kMaxValue = 10,
```